earn/src/pages/BoostPage.tsx

```diff
@@ -180,80 +184,83 @@ export default function BoostPage() {
-                          MERGE CARD INFOS
+                     CREATE UNISWAP CARD INFOS
   //////////////////////////////////////////////////////////////*/
-  const allCardInfos = useMemo(() => {
-    const merged = boostedCardInfos.concat(uniswapCardInfos);
-    return merged
-      .map(
-        (cardInfo) =>
-          new BoostCardInfo(
-            cardInfo.cardType,
-            cardInfo.nftTokenId,
-            cardInfo.uniswapPool,
-            cardInfo.currentTick,
-            cardInfo.token0,
-            cardInfo.token1,
-            cardInfo.lender0,
-            cardInfo.lender1,
-            colors.get(cardInfo.token0.logoURI) ?? cardInfo.color0,
-            colors.get(cardInfo.token1.logoURI) ?? cardInfo.color1,
-            cardInfo.position,
-            cardInfo.feesEarned,
-            cardInfo.borrower
-          )
-      )
+  const uniswapCardInfos: BoostCardInfo[] = useMemo(() => {
+    if (slot0Data === undefined || slot0Data.length !== uniswapNFTPositions.length || marketDatas === undefined) {
+      return [];
+    }
+    return uniswapNFTPositions
+      .map((position, index) => {
+        const currentTick = slot0Data[index]['tick'];
+        const marketData = marketDatas[index];
+
+        return new BoostCardInfo(
+          BoostCardType.UNISWAP_NFT,
+          position.tokenId,
+          computePoolAddress({ ...position, chainId: activeChain.id }),
+          currentTick,
+          position.token0,
+          position.token1,
+          marketData.lender0,
+          marketData.lender1,
+          colors.get(position.token0.logoURI) ?? DEFAULT_COLOR0,
+          colors.get(position.token1.logoURI) ?? DEFAULT_COLOR1,
+          position,
+          // TODO: fetch fees earned for Uniswap NFT
+          { amount0: GN.zero(position.token0.decimals), amount1: GN.zero(position.token1.decimals) },
+          null
+        );
+      })
       .filter((info) => info.lender0 !== ethers.constants.AddressZero || info.lender1 !== ethers.constants.AddressZero);
-  }, [boostedCardInfos, uniswapCardInfos, colors]);
+  }, [slot0Data, uniswapNFTPositions, marketDatas, activeChain.id, colors]);
+
+  /*//////////////////////////////////////////////////////////////
+                      CREATE BOOSTED CARD INFOS
+  //////////////////////////////////////////////////////////////*/
+  const boostedCardInfos: BoostCardInfo[] = useMemo(() => {
+    return initialBoostedCardInfos.map((info, index) => {
+      return new BoostCardInfo(
+        BoostCardType.BOOST_NFT,
+        info.nftTokenId,
+        info.uniswapPool,
+        info.currentTick,
+        info.token0,
+        info.token1,
+        info.lender0,
+        info.lender1,
+        colors.get(info.token0.logoURI) ?? info.color0,
+        colors.get(info.token1.logoURI) ?? info.color1,
+        info.position,
+        info.feesEarned,
+        info.borrower
+      );
+    });
+  }, [initialBoostedCardInfos, colors]);
```

Comment:
Bad code smell. Was masked previous by the fact that we were modifying colors in the "merge" block. I think colors should be fetched in the Uniboost file as part of fetching the rest of the card info. OR they should be passed into child components separately, so that we don't need to have `initialBoostedCardInfos` and `boostedCardInfos`